DirectX11/HackerDXGI.cpp

```diff
@@ -2062,6 +2164,296 @@ STDMETHODIMP HackerDXGISwapChain::GetLastPresentCount(THIS_
+	/* [in] */ UINT SwapChainFlags)
+{
+	LogInfo("HackerDXGISwapChain::ResizeBuffers(%s@%p) called \n", type_name(this), this);
+
+	if (G->mResolutionInfo.from == GetResolutionFrom::SWAP_CHAIN)
+	{
+		G->mResolutionInfo.width = Width;
+		G->mResolutionInfo.height = Height;
+		LogInfo("Got resolution from swap chain: %ix%i\n",
+			G->mResolutionInfo.width, G->mResolutionInfo.height);
+	}
+
+	// In Direct Mode, we need to ensure that we are keeping our 2x width backbuffer.
+	// We are specifically modifying the value passed to the call, but saving the desired
+	// resolution before this.
+	if (G->gForceStereo == 2)
+	{
+		Width *= 2;
+		LogInfo("-> forced 2x width for Direct Mode: %d \n", Width);
+	}
+
+	HRESULT hr;
+
+	if (mFakeBackBuffer) // UPSCALE_MODE 0
+	{
+		// TODO: need to consider the new code (G->gForceStereo == 2)
+		// would my stuff work this way? i guess yes. What is with the games that are not calling resize buffer
+		// just try to recreate texture with new game resolution
+		// should be possible without any issues (texture just like the swap chain should not be used at this time point)
+
+		D3D11_TEXTURE2D_DESC fd;
+		mFakeBackBuffer->GetDesc(&fd);
+
+		if (!(fd.Width == Width && fd.Height == Height))
+		{
+			mFakeBackBuffer->Release();
+
+			fd.Width = Width;
+			fd.Height = Height;
+			fd.Format = NewFormat;
+			// just recreate texture with new width and height
+			hr = mHackerDevice->CreateTexture2D(&fd, nullptr, &mFakeBackBuffer);
+		}
+		else  // nothing to resize
+			hr = S_OK;
+	}
+	else if (mFakeSwapChain) // UPSCALE_MODE 1
+	{
+		// the last parameter have to be zero to avoid the influence of the faked swap chain on the resize target function 
+		hr = mFakeSwapChain->ResizeBuffers(BufferCount, Width, Height, NewFormat, 0);
+	}
+	else
+	{
+		assert(false); // should never be triggered (class hierarchy)
+	}
+
+	if (SUCCEEDED(hr))
+	{
+		mOverlay->Resize(Width, Height);
+	}
+
+	LogInfo("  returns result = %x\n", hr);
+	return hr;
+}
+
+STDMETHODIMP HackerUpscalingDXGISwapChain::ResizeTarget(THIS_
+	/* [annotation][in] */
+	_In_  const DXGI_MODE_DESC *pNewTargetParameters)
+{
+	LogInfo("HackerUpscalingDXGISwapChain::ResizeTarget(%s@%p) called \n", type_name(this), this);
+
+	// In Direct Mode, we need to ensure that we are keeping our 2x width target.
+	if ((G->gForceStereo == 2) && (pNewTargetParameters->Width == G->mResolutionInfo.width))
+	{
+		const_cast<DXGI_MODE_DESC*>(pNewTargetParameters)->Width *= 2;
+		LogInfo("-> forced 2x width for Direct Mode: %d \n", pNewTargetParameters->Width);
+	}
+
+	/*
+		TODO: what about 3dv direct mode? why the resize target need to use double width?
+		this would resize target window and have some other drawbacks, but again
+		im not very familiar with directx 11 maybe it will cause no problems:
+		Anyway need to consider new code for the upscaling later
+	*/
+
+	// Some games like Witcher seems to drop fullscreen everytime the resizetarget is called (original one)
```

Comment:
Witcher? DX9 game. If you mean Witcher3, please use that.  It's very confusing otherwise.  More keystrokes are always better for clarity.  Or to be more accurate, too few keystrokes adds confusion and extra mental work.

I really appreciate your 'why' comments.  One of the other big value comments is a 'theory of operation' comment.  Maybe add that to the HackerDXGIUpscalingSwapChain object. Like the comments from the original Issue. No rush.